pkg/yaml/types.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2017 The Kedge Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package yaml
+
+import (
+	"encoding/json"
+	"errors"
+	"strconv"
+)
+
+// SliceArrayorIntArray represents a slice or an integer
+type SliceArrayorIntArray []string
+
+// UnmarshalJSON implements the Unmarshaller interface for the "yaml" package.
+// (since the yaml package actually used UnmarshalJSON underneath)
+// Overriding the default unmarshaller for our special one.
+func (s *SliceArrayorIntArray) UnmarshalJSON(data []byte) error {
+
+	// We create an array of strings by extracting to an interface and then iterating and figuring
+	// out each type. As per: https://golang.org/pkg/encoding/json/#Unmarshal
+	// we have a "case" for "string" and "float64"
```

Comment:
why do we have a float64 here instead int?

Reply:
Due to this, I think this is considered as valid input -
```yaml
services:
- portMappings:
  - 3306.42
```
and gets converted to -
```yaml
apiVersion: v1
kind: Service
metadata:
  creationTimestamp: null
  labels:
    app: database
  name: database
spec:
  ports:
  - name: database-3306
    port: 3306
    protocol: TCP
    targetPort: 3306
  selector:
    app: database
  type: NodePort
status:
  loadBalancer: {}
```
which is not the correct behavior, we should be throwing an error in this case :(

Reply:
agreed, and name says Int, float is not int ;-)

Reply:
See the above comment :+1: 

Yes, I understand that it says float, that is the default behaviour for JSON unmarshal: https://golang.org/pkg/encoding/json/#Unmarshal

Reply:
This will lead to a lot of errors. It has to be string or int, float can't be a valid value.

Reply:
If filtering out the float, can't be handled here, then it has to be handled somewhere else.
But doing it in the unamshaller is the right place

Reply:
I've just tried it in Kubernetes, and IntOrString from apimachinery is currently not allowing float : `for: "asdf": json: cannot unmarshal number 3.3 into Go value of type int32`